test/e2e/json-rpc/switchEthereumChain.spec.js

```diff
@@ -268,15 +310,16 @@ describe('Switch Ethereum Chain for two dapps', function () {
         await driver.delay(2000);
 
         // Switch to notification that should still be switchEthereumChain request but with a warning.
-        await switchToNotificationWindow(driver, 4);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
 
-        await driver.findElement({
-          span: 'span',
-          text: 'Switching networks will cancel all pending confirmations',
-        });
+        // THIS IS BROKEN
```

Comment:
👀

Reply:
It does seem critical for us to continue clearing pending confirmations upon switching networks. The confirmation would no longer be valid after switching. I am seeing at least one flaky test error that looks to be caused by this (here: https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/103391/workflows/9cda560b-8258-4d53-95c9-8f718b4f5473/jobs/3854013/tests)

Reply:
To be clear, switching the network does still clear pending confirmations. There is just no longer any warning to the user a that this will happen. If this was intentional UI/UX, then my label of "THIS IS BROKEN" is incorrect, but otherwise we need to find a way to add it back

Reply:
the test above was broken from flakiness around successive popup/dialogs opening and closing timings

Fixed here https://github.com/MetaMask/metamask-extension/pull/27561/commits/8bf7342e8f6510324bd6322c18a1fc92505efa5b

Reply:
I don't think the commit you linked is related to this line (this is a different test)

Reply:
Sorry I think I shared the wrong link. I meant to share this one: https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/103391/workflows/7ac38249-dd6e-4a1b-9401-f55c8273fde6/jobs/3853761/tests

Reply:
On that failure, see the fourth screenshot, the screenshot with the transaction error on it. It says "Invalid signature v value", which suggests that the chain ID was incorrect.

Either that a pending confirmation was _not_ dismissed upon the network switching, or that there was a race condition where the network switched _after_ a send transaction request was generated, but _before_ it was processed by us.

I'm guessing it's the latter. In which case, extra caution is warranted when performing actions on a dapp just before or after the network has switched. If a confirmation is "in flight" when the network is switched, or if it's triggered immediately after a switch but before the dapp has been told about the new chain ID, we'll see an error like this.